File: typescript/medium/wateringPlants.ts
function wateringPlants(plants: number[], capacity: number): number {
  let index = 0;
  let curCap = capacity;
  let rs = 0;
  for (let i = 0; i < plants.length; i++) {
    if (curCap == capacity) {
      rs += i + 1;
      rs += Math.ceil(plants[i] / capacity);
      curCap = plants[i] % capacity;
    } else {
    }
  }
}

File: java/src/medium/WateringPlants.java
package medium;

public class WateringPlants {
    public int wateringPlants(int[] plants, int capacity) {
        int curCap=capacity;
        int rs=0;
        for(int i=0;i<plants.length;i++){
            if(plants[i]<=curCap){
                curCap-=plants[i];
                rs++;
            }else{
                curCap=capacity-plants[i];
                rs+=i*2+1;
            }
        }
        return rs;
    }
}
